### ISSUE TITLE
feat(material icon): Support for Material Symbols

### ISSUE THREAD

**Issue by mark-langer** [angular/components#24845]
### Feature Description

Google just published [Material Symbols](https://material.io/blog/introducing-symbols), a fresher version of material icons that's also packaged within a variable font, giving us designers & developers more fine-grained control over the icons' looks 🎨 Plus, their default icon styles look a tad more modern (-> rounded icons).

It would be great to use them in our Angular projects through one of Angular's material components.

### Use Case

Simplify using Google's new Material Symbols font.

**Comment by thw0rted** [comment]
See also: https://developers.google.com/fonts/docs/material_symbols

**Comment by wibimaster** [comment]
I don't know how you want to simplify, but at this time, you can do that :
**index.html**: `<link href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0" rel="stylesheet">`
**your.component.ts**: `<mat-icon fontSet="material-symbols-outlined">breastfeeding</mat-icon>`

**Comment by dvalley56** [comment]
Tested on angular 13

Material symbols work flawlessly

Need to add this in index.html
```
<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,300,0,0" />
```

And `class="material-symbols-outlined"` in mat-icon

Tip: 
```
export class AppModule {
  constructor(iconRegistry: MatIconRegistry) {
    iconRegistry.setDefaultFontSetClass('material-symbols-outlined');
  }
}
```

Add this in AppModule so all mat-icons will have this class

**Comment by mmanista-bynd** [comment]
note - if applying @dvalley56's solution, you might run into the `mat-icon`s not being visible when you specify the icon type as `fontIcon` attribute of `mat-icon`, like this:
```
                  <mat-icon fontIcon="arrow_downward"></mat-icon>
```

where this:
```
                  <mat-icon>arrow_downward</mat-icon>
```
still works fine.

```
iconRegistry.setDefaultFontSetClass('material-symbols-outlined');
```
causes the `mat-ligature-font` class to be removed from the `mat-icon` elements.

A safer solution might be to only replace the `material-icons` class with `material-icons-outlined`, like this for example:
```
export class AppModule {
  constructor(private readonly iconRegistry: MatIconRegistry) {
    const defaultFontSetClasses = iconRegistry.getDefaultFontSetClass();
    const outlinedFontSetClasses = defaultFontSetClasses
      .filter((fontSetClass) => fontSetClass !== 'material-icons')
      .concat(['material-icons-outlined']);
    iconRegistry.setDefaultFontSetClass(...outlinedFontSetClasses);
  }
}
```

which keeps the `mat-ligature-font` class name and both usages of `mat-icon` work fine

**Comment by mibnd** [comment]
Hey! Great solutions. 
It seems I need to concat the 'material-symbols-outlined' string instead of the 'material-icons-outlined' in @mmanista-bynd's solution.

**Comment by ikeen0807** [comment]
There now is an NPM package availabe through which you can easily use the new set of material symbols from Google.

https://www.npmjs.com/package/material-symbols

**Comment by zarend** [comment]
Thank you for sharing, @ikeen0807 . 

@mark-langer, does material-symbols work for your use-case?

-Zach

**Comment by NoamRivlin** [comment]
with this:
`<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,300,0,0" />`

in my HTML it seems that im unable to get filled icons or symbols, only outlined.
maybe due to fill being 0?
how do you handle fill, cause SCSS doesn't help:

`.material-symbols-outlined {
    font-variation-settings: "FILL" 0, "wght" 400, "GRAD" 0, "opsz" 24;
  
    &.star-fill {
      font-variation-settings: "FILL" 1, "wght" 400, "GRAD" 0, "opsz" 24;
    }
  }`

**Comment by Spawnrad** [comment]
> with this: `<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,300,0,0" />`
> 
> in my HTML it seems that im unable to get filled icons or symbols, only outlined. maybe due to fill being 0? how do you handle fill, cause SCSS doesn't help:
> 
> `.material-symbols-outlined { font-variation-settings: "FILL" 0, "wght" 400, "GRAD" 0, "opsz" 24;
> 
> ```
> &.star-fill {
>   font-variation-settings: "FILL" 1, "wght" 400, "GRAD" 0, "opsz" 24;
> }
> ```
> 
> }`

Use this :

<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:FILL@0..1" />

You need to declare the range to be able to do : 
font-variation-settings: "FILL" 1,
or
font-variation-settings: "FILL" 0;

**Comment by ilyakonrad** [comment]
Is MatIconComponent going to support quick ways to configure icon appearance when using Material Symbols (weight, fill, grade, size)? I mean via input properties.

I think "fill" would be especially useful in projects that want to switch from Material Icons to Material Symbols, since e. g. "fill" in Material Icons Round is kind of random compared to Material Symbols where it's controlled by the user.

**Comment by mrmokwa** [comment]
Not only that, but some MD3 specs uses both styles to indicate an state.
For exemple, navigation rail within an active route, should use the filled variant while not active uses the default outlined icons.

**Comment by UnlikeMars** [comment]
[Google Fonts](https://fonts.google.com/icons?coloronly=true&icon.set=Material+Icons) now displays a note in the style selection indicating that Material Icons are deprecated.

<img width="257" alt="deprecated" src="https://github.com/user-attachments/assets/ad621bb8-4f1b-4592-8cb9-f4d02241d9b2">

**Comment by tphinney** [comment]
Yes, the Material Icons set has not gotten routine updates in at least two or three years now. Material Symbols has many more icons now as a result, and could be labeled as “current” rather than “new” at this point.

**Comment by mattiLeBlanc** [comment]
> Another way to do this besides what [@mmanista-bynd](https://github.com/mmanista-bynd) suggested is doing it via Provider. This way it runs earlier and you can also save it (the provider) as a constant in a library or anywhere to be added to your app.config.ts (newer Angular versions) or anywhere where you can add providers (e.g. main.ts in newer or older Angular versions):
> 
> {
>   provide: APP_INITIALIZER,
>   multi: true,
>   useFactory: (iconRegistry: MatIconRegistry) => () => {
>     const defaultFontSetClasses = iconRegistry.getDefaultFontSetClass();
>     const outlinedFontSetClasses = defaultFontSetClasses
>       .filter((fontSetClass) => fontSetClass !== 'material-icons')
>       .concat(['material-symbols-outlined']);
>     iconRegistry.setDefaultFontSetClass(...outlinedFontSetClasses);
>   },
>   deps: [MatIconRegistry]
> }

Thanks heaps, I was looking for this.

The Angular 19 version would be:
```
 provideAppInitializer(() => {
      const initializerFn = ((iconRegistry: MatIconRegistry) => () => {
        const defaultFontSetClasses = iconRegistry.getDefaultFontSetClass();
        const outlinedFontSetClasses = defaultFontSetClasses
          .filter((fontSetClass) => fontSetClass !== 'material-icons')
          .concat(['material-symbols-outlined']);
        iconRegistry.setDefaultFontSetClass(...outlinedFontSetClasses);
      })(inject(MatIconRegistry));
      return initializerFn();
    })
```

**Comment by coder-ashwin** [comment]
Any update on this?
https://github.com/angular/components/pull/30037 ??

**Comment by mostafabita** [comment]
To use Material Symbols in the same way you use Material Icons, simply add mat-ligature-font as the default font set class alongside your chosen Material Symbols class (e.g., material-symbols-outlined or material-symbols-rounded).

Here's how to set it up:

`matIconReg.setDefaultFontSetClass('material-symbols-outlined', 'mat-ligature-font');`

Once configured, you can use the more streamlined syntax for Material Symbols, similar to Material Icons:

`<mat-icon fontIcon="star"></mat-icon>`

This means you won't need to use `<mat-icon>star</mat-icon>` for Material Symbols anymore.

**Comment by Panossa** [comment]
> To use Material Symbols in the same way you use Material Icons, simply add mat-ligature-font as the default font set class alongside your chosen Material Symbols class (e.g., material-symbols-outlined or material-symbols-rounded).
> 
> Here's how to set it up:
> 
> `matIconReg.setDefaultFontSetClass('material-symbols-outlined', 'mat-ligature-font');`
> 
> Once configured, you can use the more streamlined syntax for Material Symbols, similar to Material Icons:
> 
> `<mat-icon fontIcon="star"></mat-icon>`
> 
> This means you won't need to use `<mat-icon>star</mat-icon>` for Material Symbols anymore.

That's actually really interesting oO
Is that something for only the newer versions?

But this issue is still relevant because at this point, Material Icons is a legacy font and should be discontinued or deprecated anyway, imo.

**Comment by mostafabita** [comment]
@Panossa I've tested this method in Angular 18 and 19, and it seems to be working. However, you're right that Angular should ideally pass Material Icons as the default icon font. Nevertheless, the method I suggested only adds a single line of style, which you could also implement without using that specific class. For example:

```
.mat-icon.material-symbols-outlined[fontIcon]::before {
    content: attr(fontIcon);
}
```

**Comment by crisbeto** [comment]
I've landed several PRs (see the linked ones above) that implement this. From 22.2 going forward all the docs refer to Material Symbols, `ng-add` will generate a link to Material Symbols and `MatIconRegistry` will attempt to detect it in order to assign the right CSS class.

**Comment by Panossa** [comment]
@crisbeto as someone who soon has to upgrade a lot of projects from v21: is there some kind of migration that would detect Material Symbols already being used and somehow work with that? Or what do I need to keep in mind when upgrading from v21 w/ Material Symbols already in use?

**Comment by crisbeto** [comment]
It's worth noting that Material Icons is still the default, because changing it would've been a massive breaking change. However, if you have a `<link>` tag with Material Symbols, we have [some logic here](https://github.com/angular/components/blob/main/src/material/icon/icon-registry.ts#L745) that will detect it and apply the correct class automatically.

So if you're on Material Icons, things will continue to work like before. If you want to switch to Symbols, all you need to do is change your `<link>` tag.

**Comment by Panossa** [comment]
I just looked at how it's done in our projects: we define a new font-face (in our case called `Material Symbols Outlined`) for the Symbols and then basically do what I wrote here:
https://github.com/angular/components/issues/24845#issuecomment-1991991731

So there's no <link> in our code, it just loads it via `@font-face` and CSS class styles. I assume, whenever Angular actually updates the icons to Symbols, I will probably be able to just remove the extra code I added to override Material Icons. Am I correct in that assumption?

**Comment by crisbeto** [comment]
Yes, you should be able to drop that JS code from your app since it happens automatically.
